File: app/src/main/java/com/gaffyhoares/data/di/ViewModelFactoryModule.java
package com.gaffyhoares.data.di;

import androidx.lifecycle.ViewModelProvider;

import com.gaffyhoares.viewmodels.ViewModelProviderFactory;

import dagger.Binds;
import dagger.Module;

@Module
public abstract class ViewModelFactoryModule {

    @Binds
    public abstract ViewModelProvider.Factory bindViewModelFactory(ViewModelProviderFactory modelProviderFactory);
}

File: app/src/main/java/com/gaffyhoares/data/di/ViewModelFactoryModule.kt
package com.gaffyhoares.data.di

import androidx.lifecycle.ViewModelProvider
import com.gaffyhoares.viewmodels.ViewModelProviderFactory
import dagger.Binds
import dagger.Module

@Module
abstract class ViewModelFactoryModule {

    @Binds
    abstract fun bindViewModelFactory(modelProviderFactory: ViewModelProviderFactory?): ViewModelProvider.Factory?
}